### PR TITLE
Make sure go.mod is correct/updated when vendor/ updates are done

### PR DESCRIPTION
**Pull request by nixpanic**
`go mod verify` does not fail when all modules under vendor/ are
updated. However it does update `go.mod` in case there is some
inconsistency.

By adding this additional verification in `make mod-check`, future
updates to vendor/ will require a correct go.mod as well.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -28,7 +28,7 @@ require (
 	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
 	google.golang.org/appengine v1.6.5 // indirect
 	google.golang.org/genproto v0.0.0-20200413115906-b5235f65be36 // indirect
-	google.golang.org/grpc v1.26.0
+	google.golang.org/grpc v1.27.0
```
Comment (Madhu-1) [inline go.mod:31]: do we need vendor directory change for this one?

**Comment by nixpanic** [comment]
On Mon, Apr 20, 2020 at 02:52:00AM -0700, Madhu Rajanna wrote:
> @Madhu-1 commented on this pull request.
> 
> 
> 
> > @@ -28,7 +28,7 @@ require (
>  	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
>  	google.golang.org/appengine v1.6.5 // indirect
>  	google.golang.org/genproto v0.0.0-20200413115906-b5235f65be36 // indirect
> -	google.golang.org/grpc v1.26.0
> +	google.golang.org/grpc v1.27.0
> 
> do we need vendor directory change for this one?

No, that has already been committed with the k8s 1.18.0 update.

**Comment by humblec** [comment]
> On Mon, Apr 20, 2020 at 02:52:00AM -0700, Madhu Rajanna wrote: @Madhu-1 commented on this pull request. > @@ -28,7 +28,7 @@ require ( golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect google.golang.org/appengine v1.6.5 // indirect google.golang.org/genproto v0.0.0-20200413115906-b5235f65be36 // indirect - google.golang.org/grpc v1.26.0 + google.golang.org/grpc v1.27.0 do we need vendor directory change for this one?
> No, that has already been committed with the k8s 1.18.0 update.

@nixpanic I dont think this change from `1.26` to `1.27` has committed with 1.18.0 update. I remember deliberately downgrading it.

**Comment by nixpanic** [comment]
@humblec hmm, is there a reason for that? `go mod verify` insists in updating the version to 1.27. Maybe there is an other/additional directive in `go.mod` needed?

**Comment by nixpanic** [comment]
By removing the line for grpc from the `go.mod` 'requires', but keeping in the 'replace' section, `go mod verify` seems to be happy.

**Comment by humblec** [comment]
> By removing the line for grpc from the `go.mod` 'requires', but

Yeah, thats what the `requires` does.  But, that upgrade can cause issue or break the stuff iic. Thats why I explictly mentoined we need `1.26.0`.

**Comment by nixpanic** [comment]
There is no direct dependency on grpc from what I can see. And keeping it in 'replace' seems to have the required effect. `go mod verify` and `go mod tidy` do not want to remove grpc, or update it this way. Do you have concerns with the current PR?

**Comment by humblec** [comment]
@nixpanic @Madhu-1 by order the vendor has to be updated following with go.mod and not vice versa, Isnt it ?

**Comment by Madhu-1** [comment]
> @nixpanic @Madhu-1 by order the vendor has to be updated following with go.mod and not vice versa, Isnt it ?

i just pulled this PR and verified we don't need vendor changes for this one

**Comment by nixpanic** [comment]
On Tue, Apr 21, 2020 at 02:13:19AM -0700, Humble Devassy Chirammal wrote:
> @nixpanic @Madhu-1 by order the vendor has to be updated following with go.mod and not vice versa, Isnt it ? 

go.mod and vendor/modules.txt should probably get updated at the same
time.

Currently vendor/modules.txt lists grpc v1.26 (just like
go.mod:replace). However go.mod:requires wants to have v1.27. This went
out of sync when the k8s 1.18.x update was merged. This current PR
brings them in sync again (grpc 1.26) and adds a `go mod verify` test
case that reports an error in case there are issues like this in the
future.

**Comment by humblec** [comment]
Fine.. LGTM.

**Comment by nixpanic** [comment]
@Madhu-1 any concerns from you?

**Comment by Madhu-1** [comment]
@MergifyIO  backport release-v2.1

**Comment by Madhu-1** [comment]
@Mergifyi backport release-v2.1

**Comment by Madhu-1** [comment]
@Mergifyio backport release-v2.1

**Pull request merged**